### DISCUSSION TITLE
clarification reggarding FreeBSD documentation in cirrus-ci.org

### DISCUSSION THREAD

**Discussion by ericoporto** [cirruslabs/cirrus-ci-docs#877] [Q&A] [answered]
I am looking at the list here: https://cirrus-ci.org/guide/FreeBSD/

on the images list there are two that I am interested:

- freebsd-13-0-snap (13.0-CURRENT)
- freebsd-13-0 (13.0-ALPHA)

What does these names means? From [FreeBSD release page here](https://www.freebsd.org/releases/13.0R/announce/) it seems 13.0 is a stable release.

Is there a base that is constant the same and stable or they constant update?

Also, what's the way to go for dependencies with FreeBSD, should I pkg install they on my pipeline, fetch sources and build or is there something similar to the Docker images we have on Windows/Linux ?

**Comment by fkorotkov** [comment, marked as answer]
Hey @ericoporto, 

At the time this documentation was last updated FreeBSD 13.0 was not release yet. Updated it in ba48ffb4fb3fd8674a3cbee347e60828ef992c19

For the second part about caching packages, unfortunately there is no feature similar to [Dockerfile as a CI environment](https://cirrus-ci.org/guide/docker-builder-vm/#dockerfile-as-a-ci-environment). But you can build a [custom VM](https://cirrus-ci.org/guide/custom-vms/) and use it. It's a bit more work but at least you'll be sure your CI environment is hermetic and packages won't magically get updated in incompatible way.

**Reply by ericoporto** [reply]
Thank you!
